code/modules/mob/living/simple_animal/hostile/bees.dm

```diff
@@ -295,29 +295,37 @@
-				user.put_in_active_hand(qb)
-				user.visible_message("<span class='notice'>[user] injects [src] with royal bee jelly, causing it to split into two bees, MORE BEES!</span>","<span class='warning'>You inject [src] with royal bee jelly, causing it to split into two bees, MORE BEES!</span>")
-			else
-				to_chat(user, "<span class='warning'>You don't have enough royal bee jelly to split a bee in two!</span>")
-		else
-			var/datum/reagent/R = GLOB.chemical_reagents_list[S.reagents.get_master_reagent_id()]
-			if(R && S.reagents.has_reagent(R.type, 5))
-				S.reagents.remove_reagent(R.type,5)
-				queen.assign_reagent(R)
-				user.visible_message("<span class='warning'>[user] injects [src]'s genome with [R.name], mutating its DNA!</span>","<span class='warning'>You inject [src]'s genome with [R.name], mutating its DNA!</span>")
-				name = queen.name
-			else
-				to_chat(user, "<span class='warning'>You don't have enough units of that chemical to modify the bee's DNA!</span>")
-	..()
+	if(!istype(I, /obj/item/reagent_containers/syringe))
+		return ..()
+
+	var/obj/item/reagent_containers/syringe/S = I
+	if(S.reagents.has_reagent(/datum/reagent/royal_bee_jelly)) //checked twice, because I really don't want royal bee jelly to be duped
+		if(!S.reagents.has_reagent(/datum/reagent/royal_bee_jelly, 5))
+			to_chat(user, "<span class='warning'>You don't have enough royal bee jelly to split a bee in two!</span>")
+			return
+
+		S.reagents.remove_reagent(/datum/reagent/royal_bee_jelly, 5)
+		var/obj/item/queen_bee/qb = new(user.drop_location())
+		qb.queen = new(qb)
+		if(queen?.beegent)
+			qb.queen.assign_reagent(queen.beegent) //Bees use the global singleton instances of reagents, so we don't need to worry about one bee being deleted and her copies losing their reagents.
+		user.put_in_active_hand(qb)
+		user.visible_message("<span class='notice'>[user] injects [src] with royal bee jelly, causing it to split into two bees, MORE BEES!</span>","<span class='warning'>You inject [src] with royal bee jelly, causing it to split into two bees, MORE BEES!</span>")
+		return
+
+	var/datum/reagent/R = GLOB.chemical_reagents_list[S.reagents.get_master_reagent_id()]
+	if(R && S.reagents.has_reagent(R.type, 5))
+		if(!(R.chemical_flags & REAGENT_CAN_BE_SYNTHESIZED)) //Adminrazine bee man bad
+			to_chat(user, "<span class='warning'>The bee writhes in anguish, are you really going to put something so foreign inside it?</span>")
+			return
+
+		S.reagents.remove_reagent(R.type,5)
+		queen.assign_reagent(R)
+		user.visible_message("<span class='warning'>[user] injects [src]'s genome with [R.name], mutating its DNA!</span>","<span class='warning'>You inject [src]'s genome with [R.name], mutating its DNA!</span>")
+		name = queen.name
+		return
+		
+	to_chat(user, "<span class='warning'>You don't have enough units of that chemical to modify the bee's DNA!</span>")
```

Comment:
Probably move this line to an early return, and have the "mutating bee DNA" be at the very bottom.